scripts.js

```diff
@@ -14,10 +16,112 @@ function createTileObject(name, top, left) {
   newTile.innerText = name;
   newTile.style.top = top + 'px';
   newTile.style.left = left + 'px';
+  newTile.id = Math.floor(Math.random() * 1000);
+  newTile.addEventListener('click', function () {
+    moveTile(this.id, this.parentElement);
+  });
+  newTile.draggable = 'true';
+  newTile.addEventListener('dragstart', function (ev) {
+    ev.dataTransfer.setData('div', ev.target.id);
+  });
+
+  tilesFirstContainer.push(newTile);
 
   return newTile;
 }
 
-function addTile() {}
+function addTile() {
+  const input = document.getElementById('tile-name').value;
+  const randomTop = Math.floor(Math.random() * (400 - 40));
+  const randomLeft = Math.floor(Math.random() * (500 - 120));
+  const tile1 = createTileObject(input, randomTop, randomLeft);
+  container1.appendChild(tile1);
+}
+
+function sortOneContainer(container, tilesToSort) {
+  tilesToSort.sort((a, b) => a.innerText.localeCompare(b.innerText));
+
+  let currentY = -40;
+  let currentX = 0;
+  let heightToAdd = 10;
+  let marginLeft = 10;
+  let marginTop = 0;
+
+  // 120 - width of tile, 30 - total padding on line
+  const maxElementsOnLine = parseInt(container.offsetWidth / (120 + 30));
```

Comment:
this method looks too complex to me. Once you have extracted the constants, think about how you can add the tiles by calculating only top and left positions for each tile.